File: src/main/java/gregtechlite/gtlitecore/client/model/WrappedModelGetter.java
package gregtechlite.gtlitecore.client.model;

import net.minecraft.client.renderer.block.model.IBakedModel;
import net.minecraft.client.renderer.block.model.ModelResourceLocation;
import net.minecraft.util.registry.IRegistry;

public interface WrappedModelGetter
{

    /**
     * A callback from the {@code ModelLoadEvent} to grad the wrapped model.
     *
     * @param modelRegistry Model registry (ModelResourceLocation, IBakedModel).
     * @return              The wrapped model.
     */
    IBakedModel getWrappedModel(IRegistry<ModelResourceLocation, IBakedModel> modelRegistry);

}

File: src/main/kotlin/gregtechlite/gtlitecore/client/model/WrappedModelGetter.kt
package gregtechlite.gtlitecore.client.model

import net.minecraft.client.renderer.block.model.IBakedModel
import net.minecraft.client.renderer.block.model.ModelResourceLocation
import net.minecraft.util.registry.IRegistry

fun interface WrappedModelGetter
{

    /**
     * A callback from the `ModelLoadEvent` to grad the wrapped model.
     *
     * @param modelRegistry The model registry `(ModelResourceLocation, IBakedModel)`.
     * @return              The wrapped model.
     */
    fun getWrappedModel(modelRegistry: IRegistry<ModelResourceLocation, IBakedModel>): IBakedModel?

}